zerver/webhooks/newrelic/view.py

```diff
@@ -58,112 +74,173 @@ def api_newrelic_webhook(
-        if info["status"].lower() == "activated":
-            content = ACTIVE_TEMPLATE.format(**info)
-        elif info["status"].lower() == "acknowledged":
-            content = DEFAULT_TEMPLATE.format(**info)
-        elif info["status"].lower() == "closed":
-            content = DEFAULT_TEMPLATE.format(**info)
-        elif info["status"].lower() == "created":
-            content = DEFAULT_TEMPLATE.format(**info)
-        else:
-            raise JsonableError(
-                _(
-                    "The newrelic webhook requires state be in [created|activated|acknowledged|closed]"
+        topic = FALL_BACK_TOPIC_TEMPLATE
+        check_send_webhook_message(request, user_profile, topic, body)
+
+    return json_success(request)
+
+
+def check_essential_fields(payload: WildValue) -> bool:
+    """
+    Checks if the essential fields are present in the payload.
+    """
+    with open("zerver/webhooks/newrelic/fixtures/essential_fields_default_payload.json") as f:
+        essential_fields = dict(json.load(f))
+        return all(key in payload for key in essential_fields)
+
+
+def parse_payload(payload: WildValue) -> Dict[str, Optional[str]]:
+    """
+    Parses the payload and turn it into a dictionary of key-value
+    pairs that can be used to format the message body.
+    """
+    priority = payload.get("priority").tame(
+        check_string_in(IncidentPrioritiesItems.as_list("name"))
+    )
+    priority_symbol = IncidentPrioritiesItems[priority].value
+    conditions_list = payload.get("alertConditionNames", ["Unknown condition"]).tame(
+        check_list(check_string)
+    )
+    conditions_str = ", ".join([f"**`{c}`**" for c in conditions_list])
+    policy_list = payload.get("alertPolicyNames", ["Unknown policy"]).tame(check_list(check_string))
+    alert_policy_str = ", ".join([f"**`{p}`**" for p in policy_list])
+
+    message_context: Dict[str, Any] = {
+        "title": payload.get("title", "No title.").tame(check_string),
+        "conditions_str": conditions_str,
+        "alert_policy_str": alert_policy_str,
+        "time_created_str": get_iso_timestamp_str(payload, IncidentTimeStamps.CREATED),
+        "time_updated_str": get_iso_timestamp_str(payload, IncidentTimeStamps.UPDATED),
+        "incident_url": payload.get("issueUrl", "https://one.newrelic.com/alerts-ai").tame(
+            check_string
+        ),
+        "total_incidents": payload["totalIncidents"].tame(check_union([check_int, check_string])),
+        "state": payload.get("state").tame(check_string_in(IncidentStates.as_list(target="name"))),
+        "status": determine_status(payload),
+        "priority": priority,
+        "priority_symbol": priority_symbol,
+    }
+
+    return message_context
+
+
+def get_iso_timestamp_str(payload: WildValue, event_type: IncidentTimeStamps) -> Optional[str]:
+    unix_time = payload.get(event_type.value, "N/A").tame(
+        check_none_or(check_union([check_int, check_string]))
```

Comment:
Since "N/A" is being set as the default ... can this be `None`?

Reply:
Yes, `check_none_or` should be safe to remove. It was there before I added "N/A" as the fallback value